src/Func.cpp

```diff
@@ -2150,16 +2181,11 @@ Func &Func::gpu_tile(VarOrRVar x, VarOrRVar y, VarOrRVar z,
 Func &Func::shader(Var x, Var y, Var c, DeviceAPI device_api) {
     invalidate_cache();
 
-    reorder(c, x, y);
-    // GLSL outputs must be stored interleaved
-    reorder_storage(c, x, y);
```

Comment:
I don't quite understand why the calls to `reorder` and `reorder_storage` were removed. Looks like `Stage::shader` reorders the loops and the storage, but we're not calling `Stage::shader` from `Func::shader`.